qemu/tests/virtio_fs_group_permission_access.py

```diff
@@ -0,0 +1,161 @@
+    """
+    add_user_cmd = params.get("add_user_cmd")
+    del_user_cmd = params.get("del_user_cmd")
+    username = params.objects("new_user")
+    windows = params.get("os_type", "windows") == "windows"
+    driver_name = params.get("driver_name")
+    fs_dest = params.get("fs_dest")
+    fs_target = params.get("fs_target")
+    cmd_run_virtiofsd = params.get("cmd_run_virtiofsd")
+
+    # nfs config
+    setup_local_nfs = params.get('setup_local_nfs')
+
+    fs_source = params.get('fs_source_dir')
+    host_session = None
+    guest_session = None
+    vm = None
+    nfs_local = None
+
+    try:
+        for _username in username:
+            cmd = add_user_cmd % _username
+            process.run(cmd)
+        user_one, user_two = username[0], username[-1]
+        # create the folder before daemon running
+        shared_dir = os.path.join("/home/" + user_one, fs_source)
+        if not os.path.exists(shared_dir):
+            process.system("runuser -l " + user_one + " -c 'mkdir -p " +
+                           shared_dir + "'")
+
+        if setup_local_nfs:
+            # delete the slash at the end
+            params["nfs_mount_dir"] = shared_dir[:-1]
+            nfs_local = nfs.Nfs(params)
+            nfs_local.setup()
+
+        # change permission of u1 home directory
+        output = process.system_output("chmod -R 777 /home/%s" % user_one)
+        error_context.context(output, test.log.info)
+        # change user u2's supplementary group to u1
+        output = process.system_output("usermod -G %s %s" % (user_one,
+                                                             user_two))
+        error_context.context(output, test.log.info)
+
+        # set fs daemon config
+        sock_path = os.path.join("/home/" + user_two,
+                                 '-'.join(('avocado-vt-vm1', 'viofs',
+                                           'virtiofsd.sock')))
+        # create the file
+        with open(sock_path, "w") as fd:
+            pass
+        params['fs_source_user_sock_path'] = sock_path
+
+        # run daemon
+        cmd_run_virtiofsd = cmd_run_virtiofsd % sock_path
+        cmd_run_virtiofsd += " --shared-dir %s" % shared_dir
+        error_context.context('Running daemon command %s' % cmd_run_virtiofsd,
+                              test.log.info)
+
+        host_session = aexpect.ShellSession("runuser -l " + user_two +
+                                            " -c '" + cmd_run_virtiofsd + "'",
+                                            auto_close=False,
+                                            output_func=utils_misc.log_line,
+                                            output_params=('virtiofs_fs-virtiofs.log',),
+                                            prompt=r"^\[.*\][\#\$]\s*$")
+        params["fs_source_base_dir"] = "/home/" + user_one
+
+        params["start_vm"] = "yes"
+        env_process.preprocess_vm(test, params, env, params["main_vm"])
```

Comment:
remove these 2 lines, and use vm.create() right after line102, so why:
1. VT has already done preprocess
2. If we do preprocess again, the cloned master image(clone_master=yes is set) will not be used for starting the VM, instead, the original system image will be used.